### DISCUSSION TITLE
How to contribute

### DISCUSSION THREAD

**Discussion by mintuhouse** [ioquatix/ioquatix#15] [General]
I am fascinated by your work with multiple repos on socketry and want to learn/ contribute in any way I can.
I am familiar with Ruby on Rails / Java Netty / NodeJS. Can you point to prerequisite resources or beginner-friendly bugs/docs?

**Comment by ioquatix** [comment]
It would be good to have more test coverage.

Gems like `protocol-http` don't have 100% test coverage.

You can see coverage by running the tests like this:

```
COVERAGE=PartialSummary bundle exec rspec
```

Please feel free to make small isolated PRs that improve test coverage. It would be a good starting place as you can learn more about the code. Note, there are a few issues with some coverage computation, so feel free to ping me if it's not making sense.

On a similar topic, code coverage by testing is good, but if you learn and understand what the code is doing, writing documentation is also useful. Follow the standard YARD style syntax, but there are a few differences. I am happy to advise the specifics on your first PR and also I can write more detailed reference for how to write documentation if that's helpful.
